accord-core/src/main/java/accord/coordinate/Coordinate.java

```diff
@@ -289,14 +168,22 @@ private void onPreAccepted()
                     }
                     else
                     {
-                        commitInvalidate(node, txnId, route, executeAt);
-                        accept(null, new Timeout(txnId, route.homeKey));
+                        node.withEpoch(executeAt.epoch, () -> {
```

Comment:
Yes, I can add some asserts. Though we will fail if we don't have them, with or without asserts.

Reply:
It's fine not to add asserts if it will fail immediately. Only reason to add asserts would be to make it fail when the wrong state/parameters are provided rather later when they are used.